### PR TITLE
DTSPO-4613 - Adding role assignment to pipeline

### PR DESCRIPTION
**Pull request by endakelly**

### CODE REVIEW

File: components/role-assignment/variables.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+variable "builtFrom" {
+  description = "Name of the GitHub repository this application is being built from."
+}
+
+variable "jenkins-mi" {
+  type = map(any)
+  default = {
+    "jenkins-cftptl-intsvc-mi"  = {},
```
Comment (timja) [inline components/role-assignment/variables.tf:16]: can you use a list? you can call `toset` on it in the for each this syntax would then be a lot nicer.

```suggestion
    default = ["jenkins-cftptl-intsvc-mi", "jenkins-cftsbox-intsvc-mi", "jenkins-ptl-mi"]
```

Comment (endakelly) [reply components/role-assignment/variables.tf]: Updated

File: components/role-assignment/role-assignment.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+data "azurerm_subscription" "subscription" {
+  provider = azurerm.dts-management-prod-intsvc
+}
+
+data "azuread_service_principal" "jenkins-mi" {
```
Comment (timja) [inline components/role-assignment/role-assignment.tf:5]:
```suggestion
data "azuread_service_principal" "jenkins" {
```

File: components/role-assignment/role-assignment.tf [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  for_each     = toset(var.jenkins-mi)
+  display_name = each.key
+}
+
+resource "azurerm_role_assignment" "jenkins-mi" {
```
Comment (timja) [inline components/role-assignment/role-assignment.tf:10]:
```suggestion
resource "azurerm_role_assignment" "jenkins_read" {
```

File: components/role-assignment/role-assignment.tf [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  for_each             = toset(var.jenkins-mi)
```
Comment (timja) [inline components/role-assignment/role-assignment.tf:11]:
```suggestion
  for_each             = toset(var.jenkins_mi)
```

File: components/role-assignment/role-assignment.tf [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  scope                = data.azurerm_subscription.subscription.id
+  role_definition_name = "Reader"
+  principal_id         = data.azuread_service_principal.jenkins-mi[each.key].id
```
Comment (timja) [inline components/role-assignment/role-assignment.tf:14]:
```suggestion
  principal_id         = data.azuread_service_principal.jenkins_mi[each.key].id
```

File: components/role-assignment/variables.tf [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+variable "product" {
+  description = "Name of the product."
+  default     = "shared-image-gallery"
+}
+
+variable "builtFrom" {
+  description = "Name of the GitHub repository this application is being built from."
+}
+
+variable "jenkins-mi" {
```
Comment (timja) [inline components/role-assignment/variables.tf:13]:
```suggestion
variable "jenkins_mi" {
```

**Review by timja — APPROVED** [review]
_ is more consistent with terraform naming, this looks much better though

**Pull request merged**